slp_tfplan/slp_tfplan/map/mapping.py

```diff
@@ -0,0 +1,164 @@
+
+    def __init__(self, trustzone: {}):
+        self.__trustzone = trustzone
+
+    @property
+    def id(self) -> str:
+        return deterministic_uuid(f"{self.type}-{self.name}")
+
+    @property
+    def type(self) -> str:
+        return self.__trustzone['type']
+
+    @property
+    def name(self) -> str:
+        return self.__trustzone['name']
+
+    @property
+    def trust_rating(self) -> int:
+        trust_rating = self.__trustzone.get('risk', {}).get('trust_rating', None)
+        return int(trust_rating) if trust_rating else None
+
+    @property
+    def is_default(self) -> bool:
+        return self.__trustzone.get('$default', False)
+
+    def __str__(self) -> str:
+        return f'{{id: {self.id}, type: {self.type}, name: {self.name}, ' \
+               f'trust_rating: {self.trust_rating}, is_default: {self.is_default}}}'
+
+
+def _exist_trustzone_by_type(trustzone_type: str, trustzones: List[TrustZoneMapping]) -> bool:
+    """
+    Returns True if exists a TrustZone with the given type, returns False otherwise
+    :param trustzone_type: The TrustZone type
+    :param trustzones: The TrustZone list
+    :return: Whether a TrustZone exists
+    """
+    return bool(list(filter(lambda tz: tz.type == trustzone_type, trustzones)))
+
+
+class AttackSurface:
+
+    def __init__(self, attack_surface: {}, trustzones: List[TrustZoneMapping]):
+        self.__attack_surface = attack_surface
+        self.__trustzones = trustzones
+        self.__validate()
+
+    def __validate(self):
+        if not self.client:
+            msg = 'Attack Surface must contain a client'
+            raise MappingFileNotValidError(MAPPING_FILE_NOT_VALID, msg, msg)
+
+        if not self.__trustzone_type or not self.__trustzones \
+                or not _exist_trustzone_by_type(self.__trustzone_type, self.__trustzones):
+            msg = 'Attack Surface must contain a valid TrustZone'
+            raise MappingFileNotValidError(MAPPING_FILE_NOT_VALID, msg, msg)
+
+    @property
+    def client(self) -> str:
+        return self.__attack_surface.get('client', None)
+
+    @property
+    def trustzone(self) -> TrustZoneMapping:
+        return next(filter(lambda tz: tz.type == self.__trustzone_type, self.__trustzones))
+
+    @property
+    def __trustzone_type(self) -> str:
+        return self.__attack_surface.get('trustzone', None)
+
+    def __str__(self) -> str:
+        return f'{{client: {self.client}, trustzone: {self.trustzone}}}'
+
+
+def _exist_default_trustzone(trustzones: List[TrustZoneMapping]):
+    return len(list(filter(lambda tz: tz.is_default, trustzones))) > 0
+
+
+class Mapping:
+
+    def __init__(self, mapping_dict: {}):
+        self.__map = mapping_dict
+        self.__validate()
+
+    def __validate(self):
+        if not self.trustzones:
+            msg = 'Mapping file must contain at least one TrustZone'
+            raise MappingFileNotValidError(MAPPING_FILE_NOT_VALID, msg, msg)
+
+        if not self.components:
```

Comment:
If there are no components, but there is a catch_all configuration, it should not raise an error.